cmd/root.go

```diff
@@ -283,7 +251,124 @@ func deployCortexManifest(repoDir string, manifestFilePath string, actionImageMa
 			}
 		}
 	}
-
+	// deploy models, experiments and runs excluding those deployed as part of campaigns
+	for _, model := range manifest.Cortex.Model {
+		v6Client, ok := cortex.(*deploy.CortexClientV6)
+		if ok {
+			relPath := parseManifestResourcePath(model)
+			// skip models deployed in campaign deployment
+			skip := false
+			for _, campaign := range campaigns {
+				if strings.HasPrefix(model, campaign) {
+					skip = true
+					break
+				}
+			}
+			if !skip {
+				if scriptTypeExists["model"] {
+					transformedResource := transformResource("model", repoDir, relPath, manifestFilePath)
+					deploy.DeployModel(*v6Client, transformedResource)
+				} else {
+					deploy.DeployModel(*v6Client, filepath.Join(repoDir, relPath))
+				}
+			}
+		} else {
+			log.Fatalln("Model deployment support is for Cortex v6 onwards")
+		}
+	}
+	for _, experiment := range manifest.Cortex.Experiment {
+		v6Client, ok := cortex.(*deploy.CortexClientV6)
+		if ok {
+			relPath := parseManifestResourcePath(experiment)
+			// skip experiment deployed in campaign deployment
+			skip := false
+			for _, campaign := range campaigns {
+				if strings.HasPrefix(experiment, campaign) {
+					skip = true
+					break
+				}
+			}
+			if !skip {
+				if scriptTypeExists["experiment"] {
+					transformedResource := transformResource("experiment", repoDir, relPath, manifestFilePath)
+					deploy.DeployExperiment(*v6Client, transformedResource)
+				} else {
+					deploy.DeployExperiment(*v6Client, filepath.Join(repoDir, relPath))
+				}
+			}
+		} else {
+			log.Fatalln("Experiment deployment support is for Cortex v6 onwards")
+		}
+	}
+	for _, run := range manifest.Cortex.Run {
+		v6Client, ok := cortex.(*deploy.CortexClientV6)
+		if ok {
+			relPath := parseManifestResourcePath(run)
+			// skip run deployed in campaign deployment
+			skip := false
+			for _, campaign := range campaigns {
+				if strings.HasPrefix(run, campaign) {
```

Comment:
could this check throw false positives?  

If I have a campaign named `test-campaign-1` and the corresponding run is called `test-campaign-1-run-xyzxyz`

this check would also pass for `test-campaign-2` ?

Reply:
nvm ignore my comment. behaviour of hasPrefix wasn't clear earlier but is clear now